### DISCUSSION TITLE
1.1.5-1.20 Release

### DISCUSSION THREAD

**Discussion by Jaffe2718** [Jaffe2718/Microphone-Text-Input#4] [General]
![icon](https://github.com/Jaffe2718/Fabric-Microphone-Text-Input/blob/1.1.5-1.20/src/main/resources/assets/mcmti/icon.png)
# What's new
- Minecraft 1.20 is supported;
- Optimize compile task.

<hr /><em>This discussion was created from the release <a href='https://github.com/Jaffe2718/Fabric-Microphone-Text-Input/releases/tag/'>1.1.5-1.20 Release</a>.</em>

**Comment by HunterYou2002** [comment]
please update to 1.21 this mod is the best thing in existence and living without it is hard

**Reply by Jaffe2718** [reply]
Sorry, I've had a busy year at work. But I might consider updating it in a month. You can also try to clone the source code and change the fabric API version and compile it yourself.

**Reply by Jaffe2718** [reply]
It is already updated, try it at https://modrinth.com/mod/microphone-text-input/version/1.1.9

**Reply by Modog500000** [reply]
Hey! I have made an api with some of the code from this mod ([VoiceLib](https://github.com/Volcano-Bay-Studios/voicelib)). It has some new features, like an API and it also automatically installs vosk on first startup. It's also multiloader for Fabric and Neoforge, however it does not include a config or the sending to chat feature as it is an api meant for developers. If you wanted I could make a PR to make this mod depend on the api and remove all the code from this one. Your choice!

**Reply by Jaffe2718** [reply]
> Hey! I have made an api with some of the code from this mod ([VoiceLib](https://github.com/Volcano-Bay-Studios/voicelib)). It has some new features, like an API and it also automatically installs vosk on first startup. It's also multiloader for Fabric and Neoforge, however it does not include a config or the sending to chat feature as it is an api meant for developers. If you wanted I could make a PR to make this mod depend on the api and remove all the code from this one. Your choice!

When I look at your API source code, it's mostly based on a tick-level event listening implementation, but it seems to register events on the logical server side, and I'm not sure if that would not work on the server side, but my mod is a pure client side mod. I'm not going to replace the API for now.

**Reply by Modog500000** [reply]
Yes I'm aware, the server side code works a bit differently, there is a method for registering listeners on the client (VoiceLibApi.registerClientSpeechListener). I was asking because I am planning on switching it to be not required on the server so that it would still stay client side. Completely up to you of course, if you want I could give you my code for downloading vosk so that users don't need to do it manually. There is still an option to change the model as well. You can check EventHandeler if this interests you.

**Reply by Modog500000** [reply]
I just got done testing to make sure, but the mod does not need to be on the server, its functions fully client sided.
